File: backend-core/src/main/java/io/github/faustofanb/admin/core/event/DomainEvent.java
package io.github.faustofanb.admin.core.event;

import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
public abstract class DomainEvent implements Serializable {
    private String eventId;
    private LocalDateTime occurredOn;
    private Long tenantId;
    private String source;

    public DomainEvent() {
        this.eventId = UUID.randomUUID().toString();
        this.occurredOn = LocalDateTime.now();
    }
}

File: src/main/kotlin/io/github/faustofan/admin/system/domain/events/DomainEvent.kt
package io.github.faustofan.admin.system.domain.events

import io.github.faustofan.admin.common.AppContextHolder
import java.time.LocalDateTime

/**
 * 领域事件基类
 */
abstract class DomainEvent {
    val occurredOn: LocalDateTime = LocalDateTime.now()
    val operatorId: Long? = AppContextHolder.get().userId
}

